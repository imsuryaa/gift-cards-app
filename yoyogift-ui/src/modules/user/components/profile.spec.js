import React from "react";
import { mount } from "enzyme";
import Profile from "./profile";

describe("Profile Test Suite", () => {
  it('should render the expected HTML', () => {
    const detailsObject = {
      email: "knowledge.com2k16@gmail.com",
      firstName: "SAS",
      lastName: "TV",
      socialProfileLink: null,
      picture: null,
      balance_points: 100000
    };
    expect(mount(<Profile detailsObject={detailsObject}/>
    ).html()).toMatchSnapshot();
  });

  it("should find a single button on render", () => {
    const detailsObject = {
      email: "knowledge.com2k16@gmail.com",
      firstName: "SAS",
      lastName: "TV",
      socialProfileLink: null,
      picture: null,
      balance_points: 100000
    };

    const wrapper = mount(<Profile detailsObject={detailsObject}/>);
    
    expect(wrapper.find('div').length).toBe(8);
    expect(wrapper.find('label').length).toBe(3);
    expect(wrapper.find('div').at(5).text()).toBe('SAS TV');
    expect(wrapper.find('div').at(6).text()).toBe('knowledge.com2k16@gmail.com');
    expect(wrapper.find('div').at(7).text()).toBe('100000 Points');
  });
})