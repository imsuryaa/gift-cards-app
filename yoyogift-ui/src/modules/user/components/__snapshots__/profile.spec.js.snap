// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Test Suite should render the expected HTML 1`] = `"<div class=\\"profileHeader\\"><img class=\\"profileImg\\" alt=\\"profile\\"></div><div class=\\"profileContent\\"><div class=\\"MuiGrid-container-1\\"><div class=\\"MuiGrid-item-2 MuiGrid-grid-xs-4-33\\"><label class=\\"label\\">Name</label><label class=\\"label\\">User Id</label><label class=\\"label\\">Balance Points</label></div><div class=\\"MuiGrid-item-2 MuiGrid-grid-xs-8-37\\"><div class=\\"MuiPaper-root-98 MuiPaper-elevation2-102 MuiPaper-rounded-99 profileValues\\">SAS TV</div><div class=\\"MuiPaper-root-98 MuiPaper-elevation2-102 MuiPaper-rounded-99 profileValues\\">knowledge.com2k16@gmail.com</div><div class=\\"MuiPaper-root-98 MuiPaper-elevation2-102 MuiPaper-rounded-99 profileValues\\">100000 Points</div></div></div></div>"`;
